File: src/javascript/get-pet.step.js
// steps/javascript/get-pet.step.js
import { get } from './js-store.js';

export const config = { type:'api', name:'JsGetPet', path:'/js/pets/:id', method:'GET', emits: [], flows: ['JsPetManagement'] };
export const handler = async (req) => {
  const pet = get(req.pathParams.id);
  return pet ? { status:200, body:pet } : { status:404, body:{ message:'Not found' } };
};

File: src/python/get_pet_step.py
# src/python/get_pet.step.py
config = { "type":"api", "name":"PyGetPet", "path":"/py/pets/:id", "method":"GET", "emits": [], "flows": ["PyPetManagement"] }

async def handler(req, _ctx=None):
    try:
        import sys
        import os
        sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..'))
        from src.services.pet_store import get
    except ImportError:
        return {"status": 500, "body": {"message": "Import error"}}
    pid = req.get("pathParams", {}).get("id")
    pet = get(pid)
    return {"status": 200, "body": pet} if pet else {"status": 404, "body": {"message": "Not found"}}
